src/document.c

```diff
@@ -788,3 +812,19 @@ DocumentField *Document_GetField(Document *d, const char *fieldName) {
   }
   return NULL;
 }
+
+const char *DocumentField_GetValueCStr(const DocumentField *df, size_t *len) {
+  *len = 0;
+  switch (df->unionType) {
+    case FLD_VAR_T_RMS:
+      return RedisModule_StringPtrLen(df->text, len);
+    case FLD_VAR_T_CSTR:
+      *len = df->strlen;
+      return df->strval;
+    case FLD_VAR_T_NUM:
+    case FLD_VAR_T_GEO:
+    case FLD_VAR_T_ARRAY:
+      break;
```

Comment:
Can you reach here? if not assert

Reply:
We should not but can be misused. An assert is a good idea. :+1: